modules/apps/dynamic-data-mapping/dynamic-data-mapping-expression/src/main/java/com/liferay/dynamic/data/mapping/expression/internal/DDMExpressionEvaluatorVisitor.java

```diff
@@ -136,6 +136,13 @@ public Object visitEqualsExpression(
 		Object object1 = visitChild(context, 0);
 		Object object2 = visitChild(context, 2);
 
+		if ((object1 instanceof Number) && (object2 instanceof Number)) {
+			BigDecimal bigDecimal1 = getBigDecimal(visitChild(context, 0));
+			BigDecimal bigDecimal2 = getBigDecimal(visitChild(context, 2));
+
+			return bigDecimal1.compareTo(bigDecimal2) == 0;
+		}
```

Comment:
@natocesarrego do you think it would be better to create a new visitor just for Numeric fields?

Reply:
Honestly, I don't think so, @mlan0.

Will you add more commits to this pull?

Reply:
Yes